[visionOS] Sign in with Apple window not anchored to the host app

rsfuller opened the pull request:
#### 5af55e89b2347fc317d8b0edaff172a7623aa38f
<pre>
[visionOS] Sign in with Apple window not anchored to the host app
<a href="https://bugs.webkit.org/show_bug.cgi?id=262230">https://bugs.webkit.org/show_bug.cgi?id=262230</a>
rdar://116142970

Reviewed by Tim Horton.

WebKit needs to pipe the callerSceneIdentifier to SOAuthorizationRequest
so that the Sign in with Apple window can be anchored correctly.

* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:
Add a delegate method to retrieve the callerSceneIdentifier on visionOS.

* Source/WebKit/UIProcess/Cocoa/SOAuthorization/SOAuthorizationSession.mm:
(WebKit::SOAuthorizationSession::continueStartAfterDecidePolicy):
Set the callerSceneIdentifier on the SOAuthorizationRequest options dict.

Canonical link: <a href="https://commits.webkit.org/268613@main">https://commits.webkit.org/268613@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fc383e142ac37a381d8965cfff93b9c79842c9f6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20166 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20601 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21226 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22066 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18819 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23842 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20763 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20279 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20386 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20290 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17524 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22916 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17459 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/24592 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18536 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18500 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22563 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19092 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/16208 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18294 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4838 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22634 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18917 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/Cocoa/SOAuthorization/SOAuthorizationSession.mm:
```diff
@@ -235,6 +235,18 @@ static bool isSameOrigin(const WebCore::ResourceRequest& request, const WebCore:
         SOAuthorizationOptionInitiatorOrigin: (NSString *)initiatorOrigin,
         SOAuthorizationOptionInitiatingAction: @(static_cast<NSInteger>(m_action))
     };
+#if PLATFORM(VISION)
```
Any /harm/ in not conditionalizing this?

rsfuller replied:
Good question, no (because the delegate method impl is already guarded), so I removed the platform guard.

rsfuller replied:
Actually, put back in a platform guard because the delegate method is only defined on IOS platforms. But at your suggestion made it more general (IOS_FAMILY). Thanks!

rsfuller committed:
[visionOS] Sign in with Apple window not anchored to the host app
https://bugs.webkit.org/show_bug.cgi?id=262230
rdar://116142970

Reviewed by Tim Horton.

WebKit needs to pipe the callerSceneIdentifier to SOAuthorizationRequest
so that the Sign in with Apple window can be anchored correctly.

* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:
Add a delegate method to retrieve the callerSceneIdentifier on visionOS.

* Source/WebKit/UIProcess/Cocoa/SOAuthorization/SOAuthorizationSession.mm:
(WebKit::SOAuthorizationSession::continueStartAfterDecidePolicy):
Set the callerSceneIdentifier on the SOAuthorizationRequest options dict.

Canonical link: https://commits.webkit.org/268613@main
